Framework/src/CheckRunner.cxx

```diff
@@ -0,0 +1,323 @@
+{
+}
+
+CheckRunner::~CheckRunner()
+{
+  // Monitoring
+  if (mCollector) {
+    std::chrono::duration<double> diff = endLastObject - startFirstObject;
+    mCollector->send({ diff.count(), "QC_checker_Time_between_first_and_last_objects_received" });
+    mCollector->send({ mTotalNumberHistosReceived, "QC_checker_Total_number_histos_treated" });
+    double rate = mTotalNumberHistosReceived / diff.count();
+    mCollector->send({ rate, "QC_checker_Rate_objects_treated_per_second_whole_run" });
+  }
+}
+
+void CheckRunner::init(framework::InitContext&)
+{
+  initDatabase();
+  initMonitoring();
+  for (auto& check : mChecks) {
+    check.init();
+  }
+}
+
+void CheckRunner::run(framework::ProcessingContext& ctx)
+{
+
+  // Save time of first object
+  if (startFirstObject == std::chrono::system_clock::time_point::min()) {
+    startFirstObject = system_clock::now();
+  }
+
+  for (const auto& input : mInputs) {
+    auto dataRef = ctx.inputs().get(input.binding.c_str());
+    if (dataRef.header != nullptr && dataRef.payload != nullptr) {
+      auto moArray = ctx.inputs().get<TObjArray*>(input.binding.c_str());
+      mLogger << "Device " << mDeviceName
+              << " received " << moArray->GetEntries()
+              << " MonitorObjects from " << input.binding
+              << AliceO2::InfoLogger::InfoLogger::endm;
+
+      for (const auto& to : *moArray) {
+        std::shared_ptr<MonitorObject> mo{ dynamic_cast<MonitorObject*>(to) };
+
+        if (mo) {
+          update(mo);
+          mTotalNumberHistosReceived++;
+        } else {
+          mLogger << "The mo is null" << AliceO2::InfoLogger::InfoLogger::endm;
+        }
+      }
+    }
+  }
+
+  // Check if compliant with policy
+  auto triggeredChecks = check(mMonitorObjects);
+  store(triggeredChecks);
+  send(triggeredChecks, ctx.outputs());
+
+  // Update global revision number
+  updateRevision();
+
+  // monitoring
+  endLastObject = system_clock::now();
+  if (timer.isTimeout()) {
+    timer.reset(1000000); // 10 s.
+    mCollector->send({ mTotalNumberHistosReceived, "objects" }, o2::monitoring::DerivedMetricMode::RATE);
+  }
+}
+
+void CheckRunner::update(std::shared_ptr<MonitorObject> mo)
+{
+  mMonitorObjects[mo->getFullName()] = mo;
+  mMonitorObjectRevision[mo->getFullName()] = mGlobalRevision;
+}
+
+std::vector<Check*> CheckRunner::check(std::map<std::string, std::shared_ptr<MonitorObject>> moMap)
+{
+  mLogger << "Running " << mChecks.size() << " checks for " << moMap.size() << " monitor objects"
+          << AliceO2::InfoLogger::InfoLogger::endm;
+
+  std::vector<Check*> triggeredChecks;
+  for (auto& check : mChecks) {
+    if (check.isReady(mMonitorObjectRevision)) {
+      auto qualityObj = check.check(moMap);
+      // Check if shared_ptr != nullptr
+      if (qualityObj) {
+        triggeredChecks.push_back(&check);
+      }
+
+      // Was checked, update latest revision
+      check.updateRevision(mGlobalRevision);
+    }
```

Comment:
I propose to add:
```
    } else {
      mLogger << "Monitor Objects for the check '" << check.getName() << "' are not ready, ignoring" << AliceO2::InfoLogger::InfoLogger::endm;
    }
```
It explains why we might see no QOs stored after "Running x checks"... or maybe "Trying to run"?